session/bootstrap.go

```diff
@@ -1228,6 +1231,14 @@ func upgradeToVer56(s Session, ver int64) {
 	doReentrantDDL(s, CreateStatsExtended)
 }
 
+func upgradeToVer57(s Session, ver int64) {
+	if ver >= version57 {
+		return
+	}
+	mustExecute(s, "DROP TABLE IF EXISTS mysql.bind_info")
+	doReentrantDDL(s, CreateBindInfoTable)
```

Comment:
This upgrade utility is pretty naive. For `TEXT` / `BLOB` type columns, if we want to build index on them, the column must be specified with a prefix length, which prevents planner from choosing PointGet / BatchPointGet for the `select ... for update` query. While according to https://docs.pingcap.com/tidb/dev/pessimistic-transaction#behaviors, only PointGet / BatchPointGet can lock the keys even if there is no record satisfying the filter. For normal IndexScan, only the records satisfying the filter would be locked, thus it is possible to have duplicate records in `mysql.bind_info` if 2 `CREATE BINDING` queries are executed simultaneously on 2 tidb instances respectively.

Therefore, in order to use PointGet / BatchPointGet, we have to change the column type of `original_sql` and `default_db` from `TEXT` to `VARCHAR`, but currently the correctness of `ALTER TABLE mysql.bind_info CHANGE original_sql VARCHAR(512)` is not guaranteed, so I simply drop this table and recreate a new one.

The side effect is that, previous bindings would be lost if users do not back up this table before upgrade. An alternative is that we can back up this table internally, i.e, save the old records before dropping `mysql.bind_info`, and insert them into the new table in this upgrade utility, but there may be potential errors if the previous `original_sql` is longer than `VARCHAR(512)`, so I don't implement this logic, at least for now.